pkg/operator/client/resource.go

```diff
@@ -111,60 +114,109 @@ func decodeManifests(manifests []string) types.Resources {
 	return resources
 }
 
-// sortResourcesForCreation sorts resources by kind based on the kind creation order.
+// groupAndSortResourcesForCreation sorts resources by kind based on the kind creation order.
 // unknown kinds are created last.
-func sortResourcesForCreation(resources types.Resources) types.Resources {
-	sortedResources := types.Resources{}
-
-	creationOrder := KindCreationOrder
-	resourcesByKind := resources.GroupByKind()
-
-	for kind := range resourcesByKind {
-		unknown := true
-		for _, creationKind := range creationOrder {
-			if kind == creationKind {
-				unknown = false
-				break
+// resources are then grouped by creation phase.
+func groupAndSortResourcesForCreation(resources types.Resources) types.Phases {
+	resourcesByPhase := resources.GroupByPhaseAnnotation(types.CreationPhaseAnnotation)
+
+	sortedPhases := getSortedPhases(resourcesByPhase)
+
+	phases := types.Phases{}
+	for _, name := range sortedPhases {
+		sortedResources := types.Resources{}
+
+		creationOrder := KindCreationOrder
+		resourcesByKind := resourcesByPhase[name].GroupByKind()
+
+		for kind := range resourcesByKind {
+			unknown := true
+			for _, creationKind := range creationOrder {
+				if kind == creationKind {
+					unknown = false
+					break
+				}
+			}
+			if unknown {
+				// unknown kinds are create last
+				creationOrder = append(creationOrder, kind)
 			}
 		}
-		if unknown {
-			// unknown kinds are create last
-			creationOrder = append(creationOrder, kind)
+
+		for _, kind := range creationOrder {
+			sortedResources = append(sortedResources, resourcesByKind[kind]...)
+		}
+
+		phase := types.Phase{
+			Name:      name,
+			Resources: sortedResources,
 		}
+		phases = append(phases, phase)
 	}
 
-	for _, kind := range creationOrder {
-		sortedResources = append(sortedResources, resourcesByKind[kind]...)
+	return phases
+}
+
+func getSortedPhases(resourcesByPhase map[string]types.Resources) []string {
```

Comment:
nit picking, how about:
```golang
phases := []int64{}
for phase := range resourcesByPhase {
  parsed, _ := strconv.ParseInt(phase, 10, 64)
  phases = append(phases, parsed)
}
sort.Slice(sortedPhases, func(i, j int) bool {
  return sortedPhases[i] < sortedPhases[j]
})
```
i feel that is easier to read and has a better performance as a plus.

Reply:
keeping this as-is so as not to change the return type